### PR TITLE
Make IE8 work

### PR DESCRIPTION
**Pull request by alanayoub**
I'm not sure if you care about IE8 but I need it to work. Feel free to merge or discard.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -523,15 +528,6 @@ var compile = function(schema, cache, root, reporter, opts) {
   validate = validate.toFunction(scope)
   validate.errors = null
 
-  validate.__defineGetter__('error', function() {
```
Comment (mafintosh) [inline index.js]: couldn't we just guard for __defineGetter__ instead?

File: index.js [hunk 2]
```diff
@@ -41,7 +41,7 @@ types.any = function() {
   return 'true'
 }
 
-types.null = function(name) {
+types[null] = function(name) {
```
Comment (mafintosh) [inline index.js]: should be types['null'] for consistency here

**Comment by mafintosh** [comment]
See my minor comments. Also could you update this so I can merge it? :)

File: index.js (hunk 1 above)
Comment (alanayoub) [inline index.js]: I haven't removed the validate.errors feature, just implemented in a different way. __defineGetter__ is non standard and has been deprecated. I can still add it back if you like?

Comment (mafintosh) [inline index.js]: Can you change it to the standard way of creating a getter then?

Comment (alanayoub) [inline index.js]: The standards compliant way would be to use `defineProperty` which isn't supported in IE<11 unfortunately. 

I opted for simply adding the error property to validate after trying a simple function that acted as a getter first. There were scope issues with that first attempt. I couldn't get validate.errors to be anything but null in this context unless I aloud for validate.errors to be called as a function.

Comment (alanayoub) [inline index.js]: If you like I can implement with `defineProperty` and remove the functionality of `validate.error` for IE versions that don't support it?

**Comment by seeden** [comment]
+1

**Comment by Reggino** [comment]
+1

**Comment by LinusU** [comment]
Since IE8 only has 0.04% global usage I don't think this is relevant anymore ☺️ 

https://caniuse.com/usage-table

**Pull request closed without merging**